### PR TITLE
fix filter extension implementation

### LINKED ISSUE #148: cql2-json filters not supported in /search GET  requests
The `filter` arg gets ignored in `/search` GET requests if `filter-lang` is `cql2-json`:
https://github.com/stac-utils/stac-fastapi-pgstac/blob/c7b17e6136d8774afec79f838705ad4ae3157bd9/stac_fastapi/pgstac/core.py#L420-L426

Without some `else` statement to handle the `filter_lang == "cql2-json"` case the `filter` parameter never gets added to `base_args` and never gets passed to the `post_request_model` via the `clean` dictionary!

From the [STAC API Filter Extension spec](https://github.com/stac-api-extensions/filter?tab=readme-ov-file)
> The implementation must support at least one of the "CQL2 Text" or "CQL2 JSON" conformance classes that define the CQL2 format used in the filter parameter:
> * CQL2 Text (http://www.opengis.net/spec/cql2/1.0/conf/cql2-text) defines that the CQL2 Text format is supported by Item Search
> * CQL2 JSON (http://www.opengis.net/spec/cql2/1.0/conf/cql2-json) defines that the CQL2 JSON format is supported by Item Search
> If both are advertised as being supported, it is only required that both be supported for GET query parameters, and that only that CQL2 JSON be supported for POST JSON requests. It is recommended that clients use CQL2 Text in GET requests and CQL2 JSON in POST requests.

I think that means that stac-fastapi-pgstac is not quite in compliance with the spec since it is not supporting cql2-json in GET requests.

### PR DESCRIPTION
**Pull request by vincentsarago**
closes #148 

This PR fixes the `CoreCrudClient.item_collection` and `CoreCrudClient.get_search` method implementation for the `filter` extension .... and add tests ;-) 

cc @hrodmn

### CODE REVIEW

**Comment by vincentsarago** [comment]
oh good point @hrodmn

I guess we need to change the type definition here https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/extensions/stac_fastapi/extensions/core/filter/request.py#L49

File: tests/resources/test_item.py [hunk 1]
```diff
@@ -1653,3 +1653,67 @@ async def test_get_filter_extension(app_client, load_test_data, load_test_collec
...
+        "/search",
+        json={
+            "filter-lang": "cql2-text",
+            "filter": f"id='{search_id}'",
+        },
+    )
+    assert resp.status_code == 200
+    fc = resp.json()
+    assert len(fc["features"]) == 1
+    assert fc["features"][0]["id"] == search_id
```
Comment (vincentsarago) [inline tests/resources/test_item.py:1719]: This is failing right now because pydantic expect a dictionary not a string

Comment (vincentsarago) [reply tests/resources/test_item.py:1719]: Oo 

https://github.com/stac-api-extensions/filter?tab=readme-ov-file#conformance-classes

> If both are advertised as being supported, it is only required that both be supported for GET query parameters, and that only that CQL2 JSON be supported for POST JSON requests. It is recommended that clients use CQL2 Text in GET requests and CQL2 JSON in POST requests.

@hrodmn I don't thin the POST request needs to support cql2-text so in fact it's a simpler fix we just need to update https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/extensions/stac_fastapi/extensions/core/filter/request.py#L73

Comment (hrodmn) [reply tests/resources/test_item.py:1719]: So we just need to make sure `filter-lang="cql2-text"` is not allowed? I realize that `cql2-text` in POST requests is not required by the spec so it's definitely not necessary but maybe convenient in some cases.

Comment (vincentsarago) [reply tests/resources/test_item.py:1719]: yes

Comment (vincentsarago) [reply tests/resources/test_item.py]: FYI: I reverted this, and removed the `cql2-text` tests for POST endpoint because it's not needed

**Pull request merged**